R/tbl_wide_summary.R

```diff
@@ -0,0 +1,195 @@
+#' Wide summary table
+#'
+#' `r lifecycle::badge("experimental")`\cr
+#' This function is similar to `tbl_summary()`, but places summary statistics
+#' wide, in separate columns.
+#' All included variables must be of the same summary type, e.g. all continuous
+#' summaries or all categorical summaries (which encompasses dichotomous variables).
+#'
+#' @inheritParams tbl_summary
+#' @param statistic (`character`)\cr
+#'   character vector of the statistics to present. Each element of the vector
+#'   will result in a column in the summary table. Default is
+#'   `c("{median}", "{p25}, {p75}")` for continuous summaries, and
+#'   `c("{n}", "{p}%")` for categorical/dichotomous summaries
+#'
+#' @return a gtsummary table of class 'tbl_wide_summary'
+#' @export
+#'
+#' @examples
+#' trial |>
+#'   tbl_wide_summary(include = c(response, grade))
+#'
+#' trial |>
+#'   tbl_strata(
+#'     strata = trt,
+#'     ~tbl_wide_summary(.x, include = c(age, marker))
+#'   )
+tbl_wide_summary <- function(data,
+                             label = NULL,
+                             statistic =
+                               switch(type[[1]],
+                                      "continuous" = c("{median}", "{p25}, {p75}"),
+                                      c("{n}", "{p}%")),
+                             digits = NULL,
+                             type = NULL,
+                             value = NULL,
+                             sort = all_categorical(FALSE) ~ "alphanumeric",
+                             include = everything()) {
+  set_cli_abort_call()
+
+  # process inputs -------------------------------------------------------------
+  check_not_missing(data)
+  check_class(data, "data.frame")
+  data <- dplyr::ungroup(data) |> .drop_missing_by_obs(by = NULL) # styler: off
+
+  cards::process_selectors(data, include = {{ include }})
+
+  cards::process_formula_selectors(data = data[include], value = value)
+
+  # assign summary type --------------------------------------------------------
+  if (!is_empty(type)) {
+    # first set default types, so selectors like `all_continuous()` can be used
+    # to recast the summary type, e.g. make all continuous type "continuous2"
+    default_types <- assign_summary_type(data, include, value)
+    # process the user-passed type argument
+    cards::process_formula_selectors(
+      data = select_prep(.list2tb(default_types, "var_type"), data[include]),
+      type = type
+    )
+    # fill in any types not specified by user
+    type <- utils::modifyList(default_types, type)
+  } else {
+    type <- assign_summary_type(data, include, value)
+  }
+
+  # summary types must be all 'continuous' OR c('categorical', 'dichotomous')
+  if (!all(unlist(type) %in% "continuous") && !all(unlist(type) %in% c("categorical", "dichotomous"))) {
+    cli::cli_abort(
+      "The summary types must all be {.val continuous} or all be {.val {c('categorical', 'dichotomous')}}.",
```

Comment:
I think either here we suggest doing something of the following sort or we allow to do the following sort:
```r
a <- tbl_wide_summary(
  trial,
  include = c(response, death, grade, stage)
)

b <- tbl_wide_summary(
  trial,
  include = c(age, marker)
)

tbl_merge(list(a, b), tab_spanner = FALSE) |>
  print()

```

Reply:
I am not 100% sure i follow the comment...but let's discuss next week!